jpeg reader and writer in io

RufaelDev opened the pull request:
Dear Point Cloud Experts,

Thank you for your great work and making it available to everyone!

My name is Rufael Mekuria, 

I have written jpeg image reader and writer classes for io that I would like to contribute back to the project. 

It is tested on windows 7 and ubuntu linux 14.4.. it uses libjpeg turbo 

I have created a unit test and added images in a folder that can be used in  these tests. 

The two classes have just two functions, 1 for file based io and one for in memory based io.

JPEG is the most used compression algorithm and relevant in 3D for compression of textures (i.e. textured mesh, point cloud etc...).

The test images can be discarded aswell beacause any jpeg image in a folder can be used for unit testing. 

looking forward to your feedback.

Best,

Rufael

VictorLamoine commented:
Hello,

```bash
CMake Warning at CMakeLists.txt:298 (find_package):
  By not providing "FindLibJPEG.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "LibJPEG", but
  CMake did not find one.

  Could not find a package configuration file provided by "LibJPEG" with any
  of the following names:

    LibJPEGConfig.cmake
    libjpeg-config.cmake

  Add the installation prefix of "LibJPEG" to CMAKE_PREFIX_PATH or set
  "LibJPEG_DIR" to a directory containing one of the above files.  If
  "LibJPEG" provides a separate development package or SDK, be sure it has
  been installed.
```

```bash
/home/victor/pcl/src/io/src/jpeg_io.cpp:233:3: warning: identifier ‘nullptr’ is a keyword in C++11 [-Wc++0x-compat]
   FILE *l_o_file = nullptr;
   ^
/home/victor/pcl/src/io/src/jpeg_io.cpp: In static member function ‘static bool pcl::io::JPEGWriter::writeJPEG(const pcl::PCLImage&, std::vector<unsigned char>&, const string&, int, bool)’:
/home/victor/pcl/src/io/src/jpeg_io.cpp:233:20: error: ‘nullptr’ was not declared in this scope
   FILE *l_o_file = nullptr;
                    ^
make[2]: *** [io/CMakeFiles/pcl_io.dir/src/jpeg_io.cpp.o] Erreur 1
```

RufaelDev commented:
Thank you for looking at this, I fixed the compiler error, I added the nullptr in a last cycle of development to avoid a compiler warning on windows which i should not have done. The code compiles now for me on ubuntu. I cannot reproduce your cmake issue on my ubuntu setup, please give me some time to test on more different machines and improve the cmake when needed.
